### PR TITLE
bugfix: global session is always begin in saga mode

### LINKED ISSUE #5045: store.mode=db 事务结束后globalsession的状态未正确修改为commited
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened

saga模式下
store.mode = db 事务结束后globalsession的状态未正确修改为commited，依然还是begin，导致global表中状态为1，等待1800s后会对事务进行回滚；
SessionHelper中endCommitted的中的
```
    public static void endCommitted(GlobalSession globalSession, boolean retryGlobal) throws TransactionException {
        if (retryGlobal || !DELAY_HANDLE_SESSION) {
            long beginTime = System.currentTimeMillis();
            boolean retryBranch = globalSession.getStatus() == GlobalStatus.CommitRetrying;
            globalSession.changeGlobalStatus(GlobalStatus.Committed);
            globalSession.end();
            if (!DELAY_HANDLE_SESSION) {
                MetricsPublisher.postSessionDoneEvent(globalSession, false, false);
            }
            MetricsPublisher.postSessionDoneEvent(globalSession, IdConstants.STATUS_VALUE_AFTER_COMMITTED_KEY, true,
                beginTime, retryBranch);
        } else {
            MetricsPublisher.postSessionDoneEvent(globalSession, false, false);
        }
    }
```
store.mode = db 时DELAY_HANDLE_SESSION为true，进不了分支，导致不能changeGlobalStatus为Committed

### Ⅳ. How to reproduce it (as minimally and precisely as possible)
把DELAY_HANDLE_SESSION的取反去掉

### Ⅵ. Environment:

- JDK version :
- Seata version:  
- OS :
- Others:

### PR DESCRIPTION
**Pull request by tuwenlin**
fixes #5045 
#5050 can not fix this bug,because in saga mode,when a global transactional end , TM report the global session status to TC rather than do commit or do rollback

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM @slievrly

**Comment by slievrly** [comment]
@tuwenlin Pls add change.

File: server/src/main/java/io/seata/server/transaction/saga/SagaCore.java [hunk 1]
```diff
@@ -194,12 +194,12 @@ public boolean doGlobalRollback(GlobalSession globalSession, boolean retrying) t
     public void doGlobalReport(GlobalSession globalSession, String xid, GlobalStatus globalStatus) throws TransactionException {
         if (GlobalStatus.Committed.equals(globalStatus)) {
             SessionHelper.removeAllBranch(globalSession, false);
-            SessionHelper.endCommitted(globalSession,false);
+            SessionHelper.endSagaGlobalSession(globalSession, GlobalStatus.Committing);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/transaction/saga/SagaCore.java:197]: why   Committed -> Committing -> Committed ?

File: server/src/main/java/io/seata/server/transaction/saga/SagaCore.java [hunk 2]
```diff
@@ -194,12 +194,12 @@ public boolean doGlobalRollback(GlobalSession globalSession, boolean retrying) t
...
             LOGGER.info("Global[{}] committed", globalSession.getXid());
         } else if (GlobalStatus.Rollbacked.equals(globalStatus)
                 || GlobalStatus.Finished.equals(globalStatus)) {
             SessionHelper.removeAllBranch(globalSession, false);
-            SessionHelper.endRollbacked(globalSession, false);
+            SessionHelper.endSagaGlobalSession(globalSession, GlobalStatus.Rollbacking);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/transaction/saga/SagaCore.java:202]: same above.

**Comment by tuwenlin** [comment]
> @tuwenlin Pls add change.

done

File: server/src/main/java/io/seata/server/transaction/saga/SagaCore.java (hunk 2 above)
Comment (tuwenlin) [reply server/src/main/java/io/seata/server/transaction/saga/SagaCore.java]: > same above.

I decide to end the GlobalSession when tc report Committed or Rollbacked

File: server/src/main/java/io/seata/server/session/SessionHelper.java [hunk 3]
```diff
@@ -201,6 +205,10 @@ public static void endRollbacked(GlobalSession globalSession, boolean retryGloba
             MetricsPublisher.postSessionDoneEvent(globalSession, IdConstants.STATUS_VALUE_AFTER_ROLLBACKED_KEY, true,
                     beginTime, retryBranch);
         } else {
+            if (globalSession.isSaga()) {
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/session/SessionHelper.java:208]: LGTM.  This PR satisfies the functional requirements, but the common method to future designs should be decoupled from the transaction mode.

**Comment by wt-better** [comment]
Saga mode can consider switch globalReport to commit/rollback，makes it consistent, reduce the judgment like `isSaga` .

**Review by slievrly — APPROVED** [review]

**Pull request merged**